### PR TITLE
expose sunset in prayerTimes

### LINKED ISSUE #105: Export sunsetTime in prayerTimes
Hi
It would be good if the sunset time was also exported from [prayerTimes ](https://github.com/batoulapps/adhan-js/blob/master/src/PrayerTimes.js) and in the [nextPrayer ](https://github.com/batoulapps/adhan-js/blob/e2465bb88f203ce7c98a0fc777181d1698d37a8a/src/PrayerTimes.js#L185) method

because in some calculation Maqrib prayer time is not same as sunset

### PR DESCRIPTION
**Pull request by jd1378**
closes  #105 
since it was remained open, I thought maybe create a pull request

### CODE REVIEW

File: Adhan.d.ts [hunk 1]
```diff
@@ -134,6 +136,7 @@ export enum Prayer {
   Sunrise,
   Dhuhr,
   Asr,
+  Sunset,
```
Comment (z3bi) [inline Adhan.d.ts:139]: we can add sunset to the PrayerTimes object but I don't want to change the `Prayer` enum.

File: src/Prayer.js [hunk 2]
```diff
@@ -3,6 +3,7 @@ const Prayer = {
     Sunrise: 'sunrise',
     Dhuhr: 'dhuhr',
     Asr: 'asr',
+    Sunset: 'sunset',
```
Comment (z3bi) [inline src/Prayer.js:6]: Should be removed as we are not changing this.

File: src/PrayerTimes.js [hunk 3]
```diff
@@ -120,13 +120,15 @@ export default class PrayerTimes {
         const sunriseAdjustment = (calculationParameters.adjustments.sunrise || 0) + (calculationParameters.methodAdjustments.sunrise || 0);
         const dhuhrAdjustment = (calculationParameters.adjustments.dhuhr || 0) + (calculationParameters.methodAdjustments.dhuhr || 0);
         const asrAdjustment = (calculationParameters.adjustments.asr || 0) + (calculationParameters.methodAdjustments.asr || 0);
+        const sunsetAdjustment = (calculationParameters.adjustments.sunset || 0) + (calculationParameters.methodAdjustments.sunset || 0);
```
Comment (z3bi) [inline src/PrayerTimes.js:123]: i thought the point was to expose the true sunset value, why would we add adjustments?

File: src/PrayerTimes.js [hunk 4]
```diff
@@ -144,6 +146,9 @@ export default class PrayerTimes {
         else if (prayer === Prayer.Asr) {
             return this.asr;
         }
+        else if (prayer === Prayer.Sunset) {
```
Comment (z3bi) [inline src/PrayerTimes.js:149]: remove

File: src/PrayerTimes.js [hunk 5]
```diff
@@ -120,13 +120,15 @@ export default class PrayerTimes {
...
         const asrAdjustment = (calculationParameters.adjustments.asr || 0) + (calculationParameters.methodAdjustments.asr || 0);
+        const sunsetAdjustment = (calculationParameters.adjustments.sunset || 0) + (calculationParameters.methodAdjustments.sunset || 0);
         const maghribAdjustment = (calculationParameters.adjustments.maghrib || 0) + (calculationParameters.methodAdjustments.maghrib || 0);
         const ishaAdjustment = (calculationParameters.adjustments.isha || 0) + (calculationParameters.methodAdjustments.isha || 0);
 
         this.fajr = roundedMinute(dateByAddingMinutes(fajrTime, fajrAdjustment), calculationParameters.rounding);
         this.sunrise = roundedMinute(dateByAddingMinutes(sunriseTime, sunriseAdjustment), calculationParameters.rounding);
         this.dhuhr = roundedMinute(dateByAddingMinutes(dhuhrTime, dhuhrAdjustment), calculationParameters.rounding);
         this.asr = roundedMinute(dateByAddingMinutes(asrTime, asrAdjustment), calculationParameters.rounding);
+        this.sunset = roundedMinute(dateByAddingMinutes(sunsetTime, sunsetAdjustment), calculationParameters.rounding);
```
Comment (z3bi) [inline src/PrayerTimes.js:131]: we should remove the call to `dateByAddingMinutes` and only do the rounding to preserve true sunset time.

File: src/PrayerTimes.js [hunk 6]
```diff
@@ -165,6 +170,9 @@ export default class PrayerTimes {
         else if (date >= this.maghrib) {
             return Prayer.Maghrib;
         }
+        else if (date >= this.sunset && this.sunset !== this.maghrib) {
```
Comment (z3bi) [inline src/PrayerTimes.js:173]: remove

File: src/PrayerTimes.js [hunk 7]
```diff
@@ -192,9 +200,12 @@ export default class PrayerTimes {
         else if (date >= this.maghrib) {
             return Prayer.Isha;
         }
-        else if (date >= this.asr) {
+        else if (date >= this.sunset) {
```
Comment (z3bi) [inline src/PrayerTimes.js:203]: revert

File: src/PrayerTimes.js [hunk 8]
```diff
@@ -192,9 +200,12 @@ export default class PrayerTimes {
...
             return Prayer.Maghrib;
         }
+        else if (date >= this.asr && this.sunset !== this.maghrib) {
```
Comment (z3bi) [inline src/PrayerTimes.js:206]: remove

**Review by z3bi — CHANGES_REQUESTED** [review]

**Comment by jd1378** [comment]
is it okay now ?

**Comment by sgtsquiggs** [comment]
This pull requests needs to be against the develop branch, which is in typescript instead of javascript.

**Pull request closed without merging**